### PR TITLE
fix: #tno-2270 - fix issues related to topic

### PR DESCRIPTION
**Pull request by kylgarmor**
- fixed issues assigning topic to content on ingest
- only content which comes from a source that is "topic enabled" will get initizalized
- event of the day folder now collects content within the last 24hr which has ANY topic
- event of the day report NEVER shows content with "Not Applicable" topic
- Topic list now shows an icon for the different topic types
- filter by typing now works on the list as well

### CODE REVIEW

File: api/net/Areas/Editor/Controllers/ContentController.cs [hunk 1]
```diff
@@ -215,7 +215,9 @@ public async Task<IActionResult> AddAsync(ContentModel model)
         newContent.OwnerId = user.Id;
         newContent.PostedOn = newContent.Status == ContentStatus.Publish || newContent.Status == ContentStatus.Published ? DateTime.UtcNow : null;
 
-        _topicScoreHelper.SetContentScore(newContent);
+        // only assign a default score to content which has a source relevent to Event of the Day
+        if (newContent.Source.UseInTopics)
+            _topicScoreHelper.SetContentScore(newContent);
```
Comment (kylgarmor) [inline api/net/Areas/Editor/Controllers/ContentController.cs:220]: only assign topics to content which comes from a source that is "topic enabled"

File: api/net/Areas/Editor/Controllers/ContentController.cs [hunk 2]
```diff
@@ -259,7 +261,9 @@ public async Task<IActionResult> UpdateAsync(ContentModel model)
             updateContent.Status == ContentStatus.Published))
             updateContent.PostedOn = DateTime.UtcNow;
 
-        _topicScoreHelper.SetContentScore(updateContent);
+        // only assign a default score to content which has a source relevent to Event of the Day
+        if ((updateContent.Source != null) && updateContent.Source.UseInTopics)
+            _topicScoreHelper.SetContentScore(updateContent);
```
Comment (kylgarmor) [inline api/net/Areas/Editor/Controllers/ContentController.cs:267]: only assign topics to content which comes from a source that is "topic enabled"

File: api/net/Areas/Editor/Controllers/ContentController.cs [hunk 3]
```diff
@@ -259,7 +261,9 @@ public async Task<IActionResult> UpdateAsync(ContentModel model)
...
```
Comment (kylgarmor) [inline api/net/Areas/Editor/Controllers/ContentController.cs:266]: only assign topics to content which comes from a source that is "topic enabled"

File: api/net/Areas/Services/Controllers/ContentController.cs [hunk 4]
```diff
@@ -180,7 +180,9 @@ public async Task<IActionResult> AddAsync(ContentModel model, int? requestorId =
             tag.Id = result.Id;
         }
 
-        _topicScoreHelper.SetContentScore(model);
+        // only assign a default score to content which has a source relevent to Event of the Day
+        if ((model.Source != null) && model.Source.UseInTopics)
+            _topicScoreHelper.SetContentScore(model);
```
Comment (kylgarmor) [inline api/net/Areas/Services/Controllers/ContentController.cs:185]: only assign topics to content which comes from a source that is "topic enabled"

File: app/editor/src/features/admin/event-of-the-day/hooks/useColumns.tsx [hunk 5]
```diff
@@ -103,6 +103,28 @@ export const useColumns = (
...
+          `option-hint ` +
+          (data.value === topicIdNotApplicable ? `type-not-applicable` : `type-${data.topicType}`)
+        }
+      >
+        {data.topicType.charAt(0)}
+      </span>
+      {data.label}
+    </div>
+  );
+
```
Comment (kylgarmor) [inline app/editor/src/features/admin/event-of-the-day/hooks/useColumns.tsx:127]: format topic option items in the list with an indicator to make it clear whether they are an Issue or Proactive topic
image

File: app/editor/src/features/admin/event-of-the-day/styled/EventOfTheDayList.tsx [hunk 6]
```diff
@@ -59,6 +59,27 @@ export const EventOfTheDayList = styled.div`
...
+  }
+  .option-hint.type-Proactive {
+    background-color: #006600;
+  }
+  .option-hint.type-Issues {
+    background-color: #bb1111;
+  }
+  .option-hint.type-not-applicable {
+    display: none;
+  }
```
Comment (kylgarmor) [inline app/editor/src/features/admin/event-of-the-day/styled/EventOfTheDayList.tsx:82]: apply styling to items in the topic list

File: libs/net/dal/Migrations/1.0.126/Up/PostUp/00-FilterAndReportSection.sql [hunk 7]
```diff
@@ -0,0 +1,223 @@
...
+        },
+        {
+          "nested": {
+            "path": "topics",
+            "query": {
+              "bool": {
+                "must": [
+                  {
+                    "match": {
+                      "topics.name": "Not Applicable"
```
Comment (kylgarmor) [inline libs/net/dal/Migrations/1.0.126/Up/PostUp/00-FilterAndReportSection.sql:50]: updated to ignore "Not Applicable" topic

File: libs/net/dal/Migrations/1.0.126/Up/PostUp/00-FilterAndReportSection.sql [hunk 8]
```diff
@@ -0,0 +1,223 @@
...
+          datasetColor = "";
+        }
+      }
+
+      @($"{{ \"label\": \"{topicType.Key}\", \"fill\": true, \"borderColor\": \"{@datasetColor}\",\"backgroundColor\": \"{@datasetColor}\", \"data\": [{@String.Join(",",topicType.Value)} ] }}")
+      if (++index < topicTypeCount) @(",")
+    }
+   ]
+}' -- template
+WHERE "name" = 'Topic Type - last 28 days';
```
Comment (kylgarmor) [inline libs/net/dal/Migrations/1.0.126/Up/PostUp/00-FilterAndReportSection.sql:137]: template is a lot more complicated now as it is grouping by day and then by topic type and adding 0 values for days where there was no content with a topic type

File: libs/net/dal/Migrations/1.0.126/Up/PostUp/00-FilterAndReportSection.sql [hunk 9]
```diff
@@ -0,0 +1,223 @@
...
+            "path": "topics",
+            "query": {
+              "exists": {
+                "field": "topics"
+              }
+            }
+          }
+        }
+      ],
+      "must_not": [
```
Comment (kylgarmor) [inline libs/net/dal/Migrations/1.0.126/Up/PostUp/00-FilterAndReportSection.sql:190]: collect all content in the last 24hrs which has a Topic

**Comment by fbarreta** [comment]
Looks good to me code wise. And as far as I'm familiar with the functionality, also looks good.

**Review by fbarreta — APPROVED** [review]

**Pull request merged**